lib/k8s/typed_resource.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module K8s
+  # generic typed resource
+  class TypedResource < Dry::Struct
+    using K8s::Util::HashDeepMerge
+
+    transform_keys do |key|
+      key = key.sub('$', '_') if key.is_a?(String)
+      key.to_sym
+    end
+    input input.strict
+
+    # @param api_path [String]
+    # @return [K8s::TypedResource, Nil]
+    def self.find_resource_class(api_path)
+      path_registry[api_path]
+    end
+
+    # @param paths [Array<String>]
+    def self.register_paths(paths)
+      paths.each do |path|
+        path_registry[path] = self
+      end
+    end
+
+    # @return [Hash]
+    def self.path_registry
+      @@path_registry ||= {}
```

Comment:
Class variables are confusing, maybe this path registry should be in `K8s::ResourceClient` or somewhere else.

Reply:
I can make better implementation once we all agree that code generation is good idea.